test/nl/surf/eduhub_rio_mapper/smoke_test.clj

```diff
@@ -55,26 +58,56 @@
                       ::ooapi/type type
                       :action      action}))))
 
+(defn req-name [request]
+  (let [action (get-in request [:headers "SOAPAction"])]
+    (if action
+      (last (str/split action #"/"))
+      (-> request :url
+          (subs (count "https://gateway.test.surfeduhub.nl/"))
+          (str/replace \/ \-)
+          (str/split #"\?")
+          first))))
+
+(defn- make-recorder [idx desc]
+  (let [mycounter (atom 0)]
+    (fn [handler request]
+      (let [response (handler request)
+            counter  (swap! mycounter inc)]
+        (let [file-name (str "test/fixtures/smoke/" idx "-" desc "/" counter "-" (req-name request) ".edn")]
+          (io/make-parents file-name)
+          (spit file-name
+                (prn-str {:request  (select-keys request [:method :headers :url :body])
```

Comment:
met headers oppassen dat we niet per ongeluk secrets loggen. wees hier expliciet welke headers gelogd worden.